Sources/AmplitudeCore/Context/AmplitudeContext.swift

```diff
@@ -4,30 +4,63 @@
 //
 //  Created by Chris Leonavicius on 3/14/25.
 //
+import Foundation
 
 @available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
 public struct AmplitudeContext {
 
+    public static var coreLibraryName: String { Constants.SDK_LIBRARY }
+    public static var coreLibraryVersion: String { Constants.SDK_VERSION }
+
     public let apiKey: String
     public let instanceName: String
     public let serverZone: ServerZone
     public let remoteConfigClient: RemoteConfigClient
     public let logger: CoreLogger
 
+    private var remoteConfigSubscription: Any?
+
+    @_spi(Internal)
+    public let diagnosticsClient: CoreDiagnostics
+
     public init(apiKey: String,
                 instanceName: String = "$default_instance",
                 serverZone: ServerZone = .US,
                 logger: CoreLogger = OSLogger(logLevel: .error)) {
+        let diagnosticsClient = DiagnosticsClient(apiKey: apiKey,
+                                                  serverZone: serverZone,
+                                                  instanceName: instanceName,
+                                                  logger: logger)
+        self.init(apiKey: apiKey,
+                  instanceName: instanceName,
+                  serverZone: serverZone,
+                  logger: logger,
+                  diagnosticsClient: diagnosticsClient)
+
+    }
+
+    @_spi(Internal)
+    public init(apiKey: String,
+                instanceName: String = "$default_instance",
+                serverZone: ServerZone = .US,
+                logger: CoreLogger = OSLogger(logLevel: .error),
+                diagnosticsClient: CoreDiagnostics) {
```

Comment:
Is there a way to get rid of the task to set RemoteConfigClient? maybe pass in a remote config client to this init as well, and set everything up externally for tests?